code/game/verbs/ooc.dm

```diff
@@ -108,12 +108,21 @@
 	var/list/messageturfs = list()//List of turfs we broadcast to.
 	var/list/messagemobs = list()//List of living mobs nearby who can hear it
 
-	for (var/turf in range(world.view, get_turf(source)))
+	for(var/turf in range(world.view, get_turf(source)))
 		messageturfs += turf
+	if(isAI(source))
+		var/mob/living/silicon/ai/AI = source
+		for(var/turf in range(world.view, get_turf(AI.eyeobj)))
+			messageturfs += turf
 
 	for(var/mob/M in player_list)
-		if (!M.client || istype(M, /mob/abstract/new_player))
+		if(!M.client || istype(M, /mob/abstract/new_player))
```

Comment:
Hmm. Looks like this check is a bit snowflake like. Also it don't cover unauthorized mob. So maybe while you are at it could do something about it? Maybe add flag or check?

Reply:
Maybe later, seems a bit out of scope to edit it now.